src/main/java/oracle/kubernetes/operator/rest/RestBackendImpl.java

```diff
@@ -474,4 +476,54 @@ private ResourceBundle getResourceBundle() {
     }
     throw new AssertionError(formatMessage(MessageKeys.RESOURCE_BUNDLE_NOT_FOUND));
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void upgradeApplications(String domainUID, UpgradeApplicationsModel appsToUpgrade ) {
+    LOGGER.entering(domainUID, "upgrade applications");
+
+    ClientHolder client = null;
+    try {
+      client = clientHelper.take();
+      upgradeApplications(client, domainUID, appsToUpgrade);
+    } finally {
+      recycleClient(clientHelper, client);
+    }
+    LOGGER.exiting();
+
+  }
+
+  private void upgradeApplications(ClientHolder client, String domainUID, UpgradeApplicationsModel appsToUpgrade) {
+    authorize(client, domainUID, Operation.update);
+
+    List<Domain> domains = getDomainsList(client);
+    Domain domain = findDomain(domainUID, domains);
+
+    String namespace = getNamespace(domainUID, domains);
+
+    // Use WLS REST API to get a list of WLS application configuration
+    Map<String, WlsAppConfig> wlsAppConfigMap = getWlsAppConfigs(client, namespace, domain, appsToUpgrade);
+    System.out.println("WlsAppConfig: " + wlsAppConfigMap);
+
+    // TBD  do later
```

Comment:
These changes aren't ready for master, where we only want to put completed work.

Reply:
Right. Just want to make sure the code related to REST API looks good from the reviews. Will continue on the TBDs.